client/common/directives/ModelFieldImage/ModelFieldImage.js

```diff
@@ -113,12 +113,14 @@ angular.module('dashboard.directives.ModelFieldImage', [
 
         scope.clear = function() {
           //Clear out an existing selected image
-          scope.data = null; //null out the data field
-          if (scope.modelData.__ModelFieldImageData && scope.modelData.__ModelFieldImageData[scope.key]) {
-            //make sure to remove any pending image uploads for this image field
-            delete scope.modelData.__ModelFieldImageData[scope.key];
+          if (confirm('Are you sure you would like to clear this photo?')) {
```

Comment:
Can you make this configurable by checking `scope.options.confirmClear`. In your project that uses the CMS you can set the `confirmClear` to `true`. Thanks!